src/Raven.Server/Documents/DatabasesLandlord.cs

```diff
@@ -168,7 +168,32 @@ private void HandleClusterDatabaseChanged(string databaseName, long index, strin
             }
         }
 
-        private void UnloadDatabase(string databaseName)
+        private void UnloadDatabase(string databaseName, bool dbRecordIsNull = false)
+        {
+            if (dbRecordIsNull)
+            {
+                UnloadDatabaseInternal(databaseName);
+                return;
+            }
+
+            if (DatabasesCache.TryGetValue(databaseName, out var databaseTask) == false)
+            {
+                // database was already unloaded or deleted
+                return;
+            }
+
+            if (databaseTask.IsCompleted == false)
+            {
+                // This case is when an unload was issued prior to the actual loading of a database.
+                databaseTask.ContinueWith(t => { UnloadDatabaseInternal(databaseName); });
```

Comment:
Here you need to check that the load itself was _successful_.
Another issue here is that here you are handling the unload in an _async_ manner.
I'm not sure that this is the right thing to do, and it may cause us concurrency issues.
In some cases, we'll issue a request to unload, but it takes time for this to be processed, but the request that was supposed to unload already returned, leading to file busy issues?

Reply:
> Here you need to check that the load itself was successful.

added